strawberryfields/engine.py

```diff
@@ -531,161 +544,542 @@ def run(self, program, *, args=None, compile_options=None, run_options=None):
         return result
 
 
-class StarshipEngine(BaseEngine):
+class RequestFailedError(Exception):
+    """Raised when a request to the remote platform returns an error response."""
+
+
+class InvalidJobOperationError(Exception):
+    """Raised when an invalid operation is performed on a job."""
+
+
+class JobFailedError(Exception):
+    """Raised when a remote job enters a 'failed' status."""
+
+
+class JobStatus(enum.Enum):
+    """Represents the status of a remote job.
+
+    This class maps a set of job statuses to the string representations returned by the
+    remote platform.
     """
-    Starship quantum program executor engine.
 
-    Executes :class:`.Program` instances on the chosen remote backend, and makes
-    the results available via :class:`.Result`.
+    OPEN = "open"
+    QUEUED = "queued"
+    CANCELLED = "cancelled"
+    COMPLETE = "complete"
+    FAILED = "failed"
+
+    @property
+    def is_terminal(self) -> bool:
```

Comment:
Just a minor suggestion here: would it make sense to go with the docstring and name this `is_final`? :slightly_smiling_face:

Reply:
Good catch, updated :+1: